packages/mobile/src/components/fields/PasswordField.tsx

```diff
@@ -14,21 +14,22 @@ export type PasswordFieldProps = PasswordInputProps & {
 
 export const PasswordField = (props: PasswordFieldProps) => {
   const { name, clearErrorOnChange = true, ...other } = props
-  const [{ value, onChange }, { touched, error }, { setError }] = useField(name)
+  const [field, { touched, error }, { setError }] = useField(name)
 
   const hasError = Boolean(touched && error)
 
   return (
     <PasswordInput
-      value={value}
+      {...field}
       error={hasError}
       helperText={hasError ? error : undefined}
       onChangeText={(e) => {
         if (clearErrorOnChange) {
           setError(undefined)
         }
-        onChange(name)(e)
+        field.onChange(name)(e)
       }}
+      onBlur={field.onBlur(name)}
```

Comment:
Some fun discoveries:

- Since we weren't handling `onBlur` before, the signup flow never set the password or confirm password fields as "touched"
- Since they weren't touched, the password completion checklist would only show green checks or empty (it only shows errors if the field has been "touched"
- Since we weren't showing errors, we never noticed we were using the wrong icon that was a whopping 48px large and had the wrong padding

After this change, the mobile sign up flow will show the red x's on the confirm password screen after touching the fields.

Reply:
nice find!